src/app/controllers/saves_controller.rb

```diff
@@ -1,19 +1,33 @@
 class SavesController < ApplicationController
-  before_action :set_user
-  before_action :set_function
+  before_action :set_function, only: %i[create index]
+  before_action :set_safe, only: :destroy
```

Comment:
[Q] Why did you use safe here instead of save?

Reply:
It's really weird but Rails thinks the singular of saves is safe

Reply:
then you just need to define an irregular inflection for it
https://api.rubyonrails.org/classes/ActiveSupport/Inflector/Inflections.html
should be in an initializer just for this